* fix intent of pointer arguments that are already associated on subroutine entry
* see https://groups.google.com/a/berkeleygw.org/forum/#!topic/help/hL_jRdFfVLo
author: Miguel Dias Costa (National University of Singapore)
--- Sigma/mtxel_cor.f90.orig	2018-03-14 08:50:52.920787000 +0800
+++ Sigma/mtxel_cor.f90	2018-03-14 08:51:28.489516380 +0800
@@ -78,11 +78,11 @@
   !> (ncoulch) Uninitialized unless freq_dep=0 .or. exact_ch=1
   SCALAR, pointer, intent(in) :: aqsch(:)
   !> (sig%ntband) Uninitialized in FF calculations
-  SCALAR, pointer, intent(out) :: acht_n1(:)
+  SCALAR, pointer, intent(inout) :: acht_n1(:)
   !> (3 or sig%nfreqeval) Uninitialized unless freq_dep=1
-  SCALAR, pointer, intent(out) :: asxt(:), acht(:)
+  SCALAR, pointer, intent(inout) :: asxt(:), acht(:)
 ! FHJ: static remainder, resolved over bands
-  SCALAR, pointer, intent(out) :: achtcor_n1(:)
+  SCALAR, pointer, intent(inout) :: achtcor_n1(:)
   SCALAR, intent(out) :: achtcor
   SCALAR, intent(out) :: asigt_imag
   integer, intent(in) :: nspin
@@ -92,9 +92,9 @@
   integer, intent(in) :: ngpown
   !> The following pointers are uninitialized unless we are running the complex version
   complex(DPC), pointer, intent(in) :: epsR(:,:,:),epsA(:,:,:) !< (neps,ngpown,sig%nFreq)
-  complex(DPC), pointer, intent(out) :: achtD_n1(:) !< (sig%ntband)
-  complex(DPC), pointer, intent(out) :: asxtDyn(:), achtDyn(:), achtDyn_cor(:), ach2tDyn(:) !< (sig%nfreqeval)
-  complex(DPC), pointer, intent(out) :: achtDyn_corb(:) !< (sig%nfreqeval)
+  complex(DPC), pointer, intent(inout) :: achtD_n1(:) !< (sig%ntband)
+  complex(DPC), pointer, intent(inout) :: asxtDyn(:), achtDyn(:), achtDyn_cor(:), ach2tDyn(:) !< (sig%nfreqeval)
+  complex(DPC), pointer, intent(inout) :: achtDyn_corb(:) !< (sig%nfreqeval)
   integer, intent(in) :: icalc
 
   SCALAR, allocatable :: epstemp(:), aqsntemp(:,:),aqsmtemp(:,:)
